presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkConfig.java

```diff
@@ -477,4 +492,34 @@ public PrestoSparkConfig setAdaptiveJoinSideSwitchingEnabled(boolean adaptiveJoi
         this.adaptiveJoinSideSwitchingEnabled = adaptiveJoinSideSwitchingEnabled;
         return this;
     }
+
+    /**
+     * Parses the catalog properties from the input string in format catalog.key=value and returns a nested map of properties.
+     */
+    public static Map<String, Map<String, String>> parseCatalogProperties(String input)
+    {
+        try {
+            Map<String, String> inputMap = MAP_SPLITTER.split(nullToEmpty(input));
+
+            Map<String, Map<String, String>> result = new HashMap<>();
+
+            for (Map.Entry<String, String> entry : inputMap.entrySet()) {
+                String key = entry.getKey();
+                String value = entry.getValue();
+
+                String[] parts = key.split("\\.");
+                if (parts.length != 2) {
+                    throw new IllegalArgumentException("Property " + key + " is not correctly formatted");
+                }
+                String catalogName = parts[0];
+                String propertyName = parts[1];
+
+                result.putIfAbsent(catalogName, new HashMap<>());
+                result.get(catalogName).put(propertyName, value);
+            }
+            return result;
+        } catch (IllegalArgumentException e) {
```

Comment:
would be better to throw a PrestoException here so the error has proper error categorization (otherwise will end up as a GENERIC_INTERNAL_ERROR).  Looks like for invalid config property errors CONFIGURATION_INVALID might be the error code to use (from StandardErrorCode.java) and for session properties it would be INVALID_SESSION_PROPERTY.

Reply:
Thanks, that's a good suggestion.